elements/lisk-chain/src/state_store/account_store.ts

```diff
@@ -64,65 +68,75 @@ export class AccountStore {
 	}
 
 	public async get<T = DefaultAsset>(
-		primaryValue: Buffer,
+		address: Buffer,
 	): Promise<Account<T>> {
 		// Account was cached previously so we can return it from memory
-		const element = this._data.get(primaryValue);
+		const cachedAccount = this._data.get(address);
 
-		if (element) {
-			return (new Account(element) as unknown) as Account<T>;
+		if (cachedAccount) {
+			return (new Account(cachedAccount) as unknown) as Account<T>;
 		}
 
 		// Account was not cached previously so we try to fetch it from db
-		const elementFromDB = await this._dataAccess.getAccountByAddress(
-			primaryValue,
+		const accountFromDB = await this._dataAccess.getAccountByAddress(
+			address,
 		);
 
 		// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-		if (elementFromDB) {
-			this._data.set(primaryValue, elementFromDB as Account);
+		if (accountFromDB) {
+			this._data.set(address, accountFromDB as Account);
+			this._initialAccountValue.set(
```

Comment:
can you create an issue to add getAccount without decoding, and decode it here?

Reply:
https://github.com/LiskHQ/lisk-sdk/issues/5428